docs/source/what_is_jupyter.md

```diff
@@ -0,0 +1,412 @@
+originally supported: Julia (ju), Python (pyt) and R (r).
+
+# How do the Jupyter notebook-editor programs work?
+
+It is common for notebook-editor programs like JupyterLab, or Jupyter
+Notebook, to share some features and workflows, because they are influenced
+by a common set of ideas about computational notebooks, their advantages,
+and how best to work with them effectively.
+
+Let's break down some of those ideas.
+
+## Interactive programming (the REPL)
+
+In the past, writing programs, running them, and seeing results was commonly
+a slower and more deliberative process than it is today.
+
+As time passed, new techniques were discovered that sped up the process between
+writing code, running it, and seeing results. Languages like Python were also
+later introduced that offered some unique advantages over older languages.
+
+For instance, Python is an interpreted language, so it does not need to be
+compiled (LINK) before it can be run (saving programmers steps and time).
+It is also a dynamically-typed language that does not require the programmer
+to specify the type of their data ahead of time before using it, which can
+sometimes save time and reduce the complexity of code (particularly for
+smaller, simpler programs).
+
+Another programming technique (which is key to the Jupyter notebook-editing
+programs) is the Read-Eval-Print-Loop (REPL), which allows a programmer to
+interactively write code, run it, keep their data and variables intact, then
+rewrite, re-run, and refine their code on-the-fly (without losing data
+after the program finishes running).
+
+A REPL is so named because the programmer writes snippets of code that are
+first read (R), then Evaluated (E) or in other words executed, the results
+are printed (P) to some kind of display or output, and that process happens
+repeatedly in a loop (L), where the REPL waits until the programmer has
+another snippet of code to execute.
+
+New snippets of code can refer to variables defined in previous Eval steps,
+because the REPL keeps objects that were created (by previous runs of the
+loop) in-memory, until the user closes their REPL.
+
+## Kernels
+
+Notebook editor programs like JupyterLab create a REPL (read more about those
+above) for each of your open notebook files, in a language of your choice. In
+the Jupyter ecosystem, each of these REPL's is called a kernel, and it holds
+the data and objects you create with your notebook code in a long-running
+program on your computer (the management and creation of these kernels
+is orchestrated by the Jupyter Server (LINK), more on that later).
+
+This is why you don't lose your data and variables when you execute multiple
+notebook cells in a row, they stay alive inside the REPL so that you can
+continue to use them to explore the data and problem you are investigating
+(though if you shut down the Jupyter server that holds the kernels, you will
+lose those variables).
```

Comment:
```suggestion
The REPL keeps your data and variables in memory when you execute multiple
notebook cells in a row. If you shut down or restart your kernel, you will
lose those variables,
and you'll need to run the code in your notebook again to get them back.
```

Again, too early to talk about servers; let's keep the kernel as the key concept in this section